src/MessageInterface.php

```diff
@@ -129,7 +131,7 @@ public function getHeaderLine($name);
      * @return static
      * @throws \InvalidArgumentException for invalid header names or values.
      */
-    public function withHeader($name, $value);
+    public function withHeader(string $name, $value);
```

Comment:
This one is interesting; when we bump to PHP 8, we could make `$value` into `string|array`, though an array type would still require validation.

Reply:
So should we go ahead and do that before the vote?  It seems wise to do so if that's what the spec already calls for.

Reply:
No - we can do it for a later minor version that bumps the minimum version to 8. For this one, we should keep PHP 7 support (per my other comments).